nix/shell-plugins.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, config, is-home-manager, ... }:
+with lib;
```

Comment:
`with lib;` brings library functions into scope like `map`, `baseNameOf`, `getExe` etc.

They could also be referenced without this line as `lib.map`, `lib.baseNameOf`, `lib.getExe`, for example.